### ISSUE TITLE
WithFilesOnly GlobOption - Returns matching files, but not matching directories

### ISSUE THREAD

**Issue by djaglowski** [bmatcuk/doublestar#79]
Is it possible to configure `Glob` in such a way that it will return matching files but not matching directories?

For example, supposing that I have no knowledge of naming or structure, I just want to find all _files_ immediately in the `/var/log` directory.
```
/var/log/some_file
/var/log/another_file
/var/log/some_dir/foo
```

If I specify the pattern `/var/log/*`, the result is `[ "some_file", "another_file", "some_dir" ]`. 

Is there a way to match only `[ "some_file", "another_file" ]`? If I'm understanding correctly, this is not currently possible. If that is the case, would you be open to adding a new `GlobOption` that specifies whether the result should include only files?

For context, this library is being used in the OpenTelemetry Collector, primarily for [finding log files](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/stanza/fileconsumer/finder.go).

**Comment by bmatcuk** [comment]
Hey @djaglowski,

I've added a `WithFilesOnly` option in [this PR](https://github.com/bmatcuk/doublestar/pull/81) - do you mind giving it a try to make sure it works before I merge it?

**Comment by djaglowski** [comment]
Hi @bmatcuk, Thanks so considering my request and putting together a PR so quickly. I've tested the changes [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/17362), and everything looks good to me.

**Comment by bmatcuk** [comment]
Awesome! I just released v4.5.0 with this PR merged.
